code/tailer/datastripe.q

```diff
@@ -16,45 +16,64 @@ modaccess:{[accesstab]};
     .lg.o[`reload;"reload command has been called remotely"];
 
     // remove periods of data from tables
-    t:tables[`.] except .wdb.ignorelist;
     lasttime:nextp-.ds.periodstokeep*(nextp-currp);
 
     // update the access table in the wdb
     // on first save down we need to replace the null valued start time in the access table
     // using the first value in the saved data
-    starttimes:.ds.getstarttime each t;
+    starttimes:.ds.getstarttime each .wdb.tablelist[];
     .ds.access:update start:starttimes^start, end:?[(nextp>starttimes)&(starttimes<>0Np);nextp;0Np], stptime:data[][`p] from .ds.access;
     modaccess[.ds.access];
 
     // call the savedown function
-    .ds.savealltablesoverperiod[.ds.td;nextp;lasttime];
-    .lg.o[`reload;"Kept ",string[.ds.periodstokeep]," period",$[.ds.periodstokeep>1;"s";""]," of data from : ",", " sv string[t]];
+    .ds.savealltablesoverperiod[.ds.td;lasttime];
+    .lg.o[`reload;"Kept ",string[.ds.periodstokeep]," period",$[.ds.periodstokeep>1;"s";""]," of data from : ",", " sv string[.wdb.tablelist[]]];
     
     // update the access table on disk
-    atab:get ` sv(.ds.td;.proc.procname;`access);
+    accesspath: ` sv(.ds.td;.proc.procname;`$ string .wdb.currentpartition;`access);
+    atab:get accesspath;
     atab,:() xkey .ds.access;
-    (` sv(.ds.td;.proc.procname;`access)) set atab;
+    accesspath set atab;
 
     // update the access table in the gateway
     handles:(.servers.getservers[`proctype;`gateway;()!();1b;1b])[`w];
     .ds.updategw each handles;
 
     };
 
+.wdb.datastripeendofday:{[pt;processdata]
+    //save all tables
+    .ds.savetables[.ds.td;] each .wdb.tablelist[];
+    //clear tables
+    @[`.;;0#] each .wdb.tablelist[];
+    //move to next partition
+    .wdb.currentpartition:pt+1;
+    //create accesspath
+    accesspath: ` sv(.ds.td;.proc.procname;`$ string .wdb.currentpartition;`access);
+    //make access for next partition
+    .ds.access:([]table:.wdb.tablelist[]; start:0Np; end:0Np; stptime:0Np; keycol:`sym^.wdb.tablekeycols .wdb.tablelist[]);
+    modaccess[.ds.access];
+    accesspath set .ds.access;
+    };
+
 initdatastripe:{
     // update endofperiod function
     endofperiod::.wdb.datastripeendofperiod;
     
+    //update endofday function
+    endofday::.wdb.datastripeendofday;
```

Comment:
could we remove this update endofday? I've created an endofday function in tailer.q that will call .wdb.datastripeendofday so this would stop that from running

Reply:
That would work with your code, but why do you have an end of day function in the tailer that calls an end of day function?

Reply:
would you not just reassign it within datastripe.q the same way eop is

Reply:
at end of day the tailer needs to send a message to the sort process to move the data to hdb and going off the wdb code there was an endofday and endofdaysave - your function eliminates the need for an endofdaysave so I was calling that in endofday. but whatever setup would work best, we just need to integrate both

Reply:
Alright no worries, that makes sense, yeh calling it from your tailer eod makes sense.